java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValue.java

```diff
@@ -105,25 +96,8 @@ public boolean references(SymbolicValue other) {
   }
 
   @Override
-  public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    SymbolicValue that = (SymbolicValue) o;
-    return id == that.id;
-  }
-
-  @Override
-  public int hashCode() {
-    return id;
-  }
-
-  @Override
   public String toString() {
-    return "SV_" + id;
+    return "SV_" + hashCode() % 100;
```

Comment:
Why the `%100` ? Only to reduce the size of the number? I think we should find something cleaner, like for instance only displaying the first 6-8 digits (a bit like commits id on github)